xds/src/main/java/io/grpc/xds/internal/sds/trust/CertificateUtils.java

```diff
@@ -61,5 +61,17 @@ private static synchronized void initInstance() throws CertificateException {
     }
   }
 
+  /** See {@link CertificateFactory#generateCertificate(InputStream)} enerateCertificate}. */
+  public static synchronized X509Certificate toX509Certificate(InputStream inputStream)
+          throws CertificateException, IOException {
+    initInstance();
+    try {
+      Certificate cert = factory.generateCertificate(inputStream);
+      return (X509Certificate)cert;
+    } finally {
+      inputStream.close();
```

Comment:
Seems we can use java7's try-with-resource now.

Reply:
added the try-with-resource in the callers